modules/stochastic_tools/src/actions/ParameterStudyAction.C

```diff
@@ -0,0 +1,711 @@
+      auto & outputs = params.set<std::vector<OutputName>>("outputs");
+      if (qoi_output.contains("csv"))
+        outputs.push_back(outputName("csv"));
+      if (qoi_output.contains("json"))
+        outputs.push_back(outputName("json"));
+      if (qoi_output.contains("none"))
+        outputs = {"none"};
+      params.set<ExecFlagEnum>("execute_on") = {EXEC_TIMESTEP_END};
+      _problem->addReporter("StochasticReporter", stochasticReporterName(), params);
+      if (_show_objects)
+        showObject("StochasticReporter", stochasticReporterName(), params);
+    }
+
+    // Add statistics object
+    if (_compute_stats)
+    {
+      auto params = _factory.getValidParams("StatisticsReporter");
+      auto & reps = params.set<std::vector<ReporterName>>("reporters");
+      for (const auto & qoi : getParam<std::vector<ReporterName>>("quantities_of_interest"))
+        reps.push_back(quantityOfInterestName(qoi));
+      params.set<MultiMooseEnum>("compute") = "mean stddev";
+      params.set<MooseEnum>("ci_method") = "percentile";
+      params.set<std::vector<Real>>("ci_levels") = {0.01, 0.05, 0.1, 0.9, 0.95, 0.99};
+      params.set<unsigned int>("ci_replicates") = 1000;
+      params.set<ExecFlagEnum>("execute_on") = {EXEC_TIMESTEP_END};
+      params.set<std::vector<OutputName>>("outputs") = {outputName("json")};
+      _problem->addReporter("StatisticsReporter", statisticsName(), params);
+      if (_show_objects)
+        showObject("StatisticsReporter", statisticsName(), params);
+    }
+  }
+  else if (_current_task == "add_control")
+  {
+    // Add command-line control if the multiapp mode warrants it
+    if (_multiapp_mode <= 1)
+    {
+      auto params = _factory.getValidParams("MultiAppSamplerControl");
+      params.set<MultiAppName>("multi_app") = multiappName();
+      params.set<SamplerName>("sampler") = samplerName();
+      params.set<std::vector<std::string>>("param_names") =
+          getParam<std::vector<std::string>>("parameters");
+      auto control =
+          _factory.create<Control>("MultiAppSamplerControl", multiappControlName(), params);
+      _problem->getControlWarehouse().addObject(control);
+      if (_show_objects)
+        showObject("MultiAppSamplerControl", multiappControlName(), params);
+    }
+  }
+}
+
+DistributionName
+ParameterStudyAction::distributionName(unsigned int count) const
+{
+  return "study_distribution_" + std::to_string(count);
+}
+
+std::vector<DistributionName>
+ParameterStudyAction::distributionNames(unsigned int full_count) const
+{
+  std::vector<DistributionName> dist_names;
+  for (const auto & i : make_range(full_count))
+    dist_names.push_back(distributionName(i));
+  return dist_names;
+}
+
+ReporterName
+ParameterStudyAction::quantityOfInterestName(const ReporterName & qoi) const
+{
+  return ReporterName(stochasticReporterName(), qoi.getObjectName() + ":" + qoi.getValueName());
+}
+
+void
+ParameterStudyAction::showObject(std::string type,
```

Comment:
This could be nice util function in the framework. I think @GiudGiud also worked on a verbose output from actions.

Reply:
I'm not sure this general enough for a util. I think there is already something somewhere else, but it was way too verbose for my liking. @GiudGiud what do you think?

Reply:
I have not worked on verbose outputs from action.
How does this compare to show_actions?
Can you paste the expected log somewhere?

Reply:
`show_actions` just outputs the task and action, not what is actually being built.

This is what `showObject` looks like:
![image](https://user-images.githubusercontent.com/12819067/206252433-d795e698-6132-4673-940c-ecb0787472c4.png)

Reply:
Looks nice. 
We do have some dump option that will dump the parameters to a file.
I dont know about having it in the log though

Reply:
So should this function be put in the Action.h that child classes can call to display recently created objects?

Reply:
Or maybe something in FEProblemBase or the warehouse?

Reply:
Not sure.
What is the motivation for that?
You want to make it open for every app rather than just STM?

Reply:
It was something @grmnptr suggested since it is general and not specific to this action. I was nice for debugging during development, but I don't see it being that useful for users (pun intended).